os/pm/pm_checkstate.c

```diff
@@ -127,37 +121,32 @@ enum pm_state_e pm_checkstate(int domain)
 	 */
 
 	now = clock_systimer();
-	if (now - pdom->stime >= TIME_SLICE_TICKS) {
-		int16_t accum;
+	if (now - g_pmglobals.stime >= TIME_SLICE_TICKS) {
 
-		/* Sample the count, reset the time and count, and assess the PM
-		 * state.  This is an atomic operation because interrupts are
-		 * still disabled.
+		/* Reset the time and recommended board to sleep.
+		 * This is an atomic operation because interrupts are still disabled.
 		 */
 
-		accum       = pdom->accum;
-		pdom->stime = now;
-		pdom->accum = 0;
+		g_pmglobals.stime = now;
+		g_pmglobals.recommended = PM_SLEEP;
 
-		/* Reassessing the PM state may require some computation.  However,
-		 * the work will actually be performed on a worker thread at a user-
-		 * controlled priority.
-		 */
-
-		(void)pm_update(domain, accum);
 	}
```

Comment:
in pm_update , we were able to change the state based on time threshold , but now there is no way to do that . In pm_update , we were able to set threshold count for each state transition . But now its not there